### ISSUE TITLE
afterFinish() is not called on a dry-run

### ISSUE THREAD

**Issue by lukeredpath** [Ahmed-Ali/Cucumberish#91]
Is there a reason for this behaviour?

I'm running the steps in dry-run mode to count the number of features with a given tag, using `beforeTagged()` to increment a counter. I only want to run this in dry run mode. I want to use `afterFinish()` to print the total but it doesn't get called.

**Comment by rafaelclaycon** [comment]
What do you mean by "dry-run"?

I'm running into an issue that I believe could be related which is the `beforeTagged()` not being called when I run a single test putting its tag on the `includeTags`: in `Cucumberish.executeFeatures`.

**Comment by lukeredpath** [comment]
@rafaelclaycon I have absolutely no idea what I meant by "dry-run" given as I posted this over 18 months ago. ;)

**Comment by rafaelclaycon** [comment]
No problem, have a good one.

**Comment by olyv** [comment]
In other Cucumber ports 'dry-run' is used to validate feature files syntax and find missing steps without actual execution of features. And I wonder if it's available in Cucumberish ;)

**Comment by rafaelclaycon** [comment]
@olyv This setting did not work as expected for me. I deleted some steps from my tests and ran them with that option set. The tests were not actually executed, as you said, but the steps I removed were not reported missing. :(

**Comment by olyv** [comment]
For it works completely opposite way: it shows all steps as undefined despite the fact I have step definitions 😆 

```
Dry Run Results
=====================================================
Found the following undefined steps:

all my steps are listed below  
```

**Comment by olyv** [comment]
yes, my steps start with capital letter. Never tried lowercase first letter
